src/overlays/actors/ovl_Demo_Kankyo/z_demo_kankyo.c

```diff
@@ -30,51 +51,951 @@ const ActorInit Demo_Kankyo_InitVars = {
+                    D_8098CF80 = 1;
+                    sRainScale = 5;
+                    break;
+                default:
+                    Actor_Kill(&this->actor);
+                    break;
+            }
+            break;
+        case DEMOKANKYO_ROCK_1:
+        case DEMOKANKYO_ROCK_2:
+        case DEMOKANKYO_ROCK_3:
+        case DEMOKANKYO_ROCK_4:
+        case DEMOKANKYO_ROCK_5:
+            globalCtx->roomCtx.curRoom.segment = NULL;
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = Rand_ZeroOne() * 0.5f + 0.5f;
+            this->unk_150[0].unk_0.x = Rand_ZeroOne() * 3.0f + 1.0f;
+            this->unk_150[0].unk_0.y = Rand_ZeroOne() * 3.0f + 1.0f;
+            this->unk_150[0].unk_0.z = Rand_ZeroOne() * 3.0f + 1.0f;
+            break;
+        case DEMOKANKYO_CLOUDS:
+            for (i = 0; i < 30; i++) {
+                this->unk_150[i].unk_20 = Rand_ZeroOne() * 65535.0f;
+                this->unk_150[i].unk_18 = Rand_ZeroOne() * 100.0f + 60.0f;
+            }
+            break;
+        case DEMOKANKYO_DOOR_OF_TIME:
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 1.0f;
+            this->unk_150[0].unk_18 = 0.0f;
+            if (!(gSaveContext.eventChkInf[4] & 0x800)) {
+                Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_TOKI,
+                                   this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0,
+                                   0, 0x0000);
+            } else {
+                globalCtx->unk_11D30[1] = 0xFF;
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case DEMOKANKYO_LIGHT_PLANE:
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 1.0f;
+            this->unk_150[0].unk_18 = 0.0f;
+            break;
+        case DEMOKANKYO_WARP_OUT:
+        case DEMOKANKYO_WARP_IN:
+            Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_ITEMACTION);
+            this->actor.flags |= 0x2000000;
+            this->actor.room = -1;
+            this->warpTimer = 35;
+            this->sparkleCounter = 0;
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 1.0f;
+            if (this->actor.params == DEMOKANKYO_WARP_OUT) {
+                Audio_PlaySoundGeneral(NA_SE_EV_SARIA_MELODY, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            }
+            break;
+        case DEMOKANKYO_SPARKLES:
+            this->warpTimer = 35;
+            this->sparkleCounter = 0;
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 1.0f;
+            break;
+        default:
+            break;
+    }
+    for (i = 0; i < 30; i++) {
+        this->unk_150[i].unk_22 = 0;
+    }
+    DemoKankyo_SetupAction(this, DemoKankyo_SetupType);
+}
+
+void DemoKankyo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    if (thisx) {}
```

Comment:
Any chance this can match with something better, like declaring `this` even tho it's not used (might need something with `globalCtx2` as well)

Reply:
Unfortunately just declaring something like that doesn't match, also tried declaring both.